tensorflow/python/kernel_tests/relu_op_test.py

```diff
@@ -216,6 +219,18 @@ def loss():
     self.evaluate(optimizer.minimize(loss))
     self.assertAllClose(x.read_value(), 50.0)
 
+  def testGradientNoElement(self):
+    with self.cached_session():
+
+      def f(x):
+        with backprop.GradientTape() as tape:
+          tape.watch(x)
+          y = nn_ops.relu(x)
+        return tape.gradient(y, x)
+
+      x = np.asarray([[],[]], dtype=np.float32)
+      z = gradient_checker_v2.compute_gradient(f, [x])[0][0]
+      self.assertAllClose(z, np.reshape(x, (0, 0)))
```

Comment:
Since there is nothing to compare, I feel we can be strict and use Equal instead of Close. But I'm fine with using Close too. What do you think? If you prefer not to change, please let me know and I'll approve the PR.
```suggestion
      self.assertAllEqual(z, np.reshape(x, (0, 0)))
```

Reply:
@penpornk Thanks for you comments, I think use assertAllEqual is better, change it to assertAllEqual.